src/test/groovy/graphql/nadel/NadelE2ETest.groovy

```diff
@@ -50,9 +50,108 @@ class NadelE2ETest extends Specification {
         def result = nadel.execute(nadelExecutionInput)
 
         then:
-        1 * delegatedExecution.delegate(_) >> completedFuture(delegatedExecutionResult)
+        1 * delegatedExecution.delegate(_) >> { args ->
+            DelegatedExecutionParameters params = args[0]
+            assert AstPrinter.printAstCompact(params.query) == "query {hello {name}}"
+            completedFuture(delegatedExecutionResult)
+        }
+        result.get().data == data
+    }
+
+    def "query with rename and alias"() {
+
+        DelegatedExecution delegatedExecution = Mock(DelegatedExecution)
+        ServiceDataFactory serviceDataFactory = new ServiceDataFactory() {
+            @Override
+            DelegatedExecution getDelegatedExecution(String serviceName) {
+                return delegatedExecution
+            }
+
+            @Override
+            GraphQLSchema getPrivateSchema(String serviceName) {
+            }
+        }
+        def nsdl = '''
+         service MyService {
+            type Query{
+                hello: World   
+            } 
+            type World {
+                name: String <= $source.name2
+            }
+         }
+        '''
+        def query = """
+        { hello {title: name}}
+        """
+        given:
+        Nadel nadel = newNadel()
+                .dsl(nsdl)
+                .serviceDataFactory(serviceDataFactory)
+                .build()
+        NadelExecutionInput nadelExecutionInput = newNadelExecutionInput()
+                .query(query)
+                .build()
+        def data = [hello: [title: "earth"]]
+        DelegatedExecutionResult delegatedExecutionResult = new DelegatedExecutionResult(data)
+        when:
+        def result = nadel.execute(nadelExecutionInput)
+
+        then:
+        1 * delegatedExecution.delegate(_) >> { args ->
+            DelegatedExecutionParameters params = args[0]
+            assert AstPrinter.printAstCompact(params.query) == "query {hello {title:name2}}"
+            completedFuture(delegatedExecutionResult)
+        }
         result.get().data == data
+    }
+
+    def "query with named fragments"() {
```

Comment:
I think we should limit the e2e tests we have and focus on the unit test. So I think one of the two is enough, or maybe merge them into one.